File: c_sharp/Enumeration/Enumeration/BloodGroup.cs
﻿using System.ComponentModel.DataAnnotations;

namespace PeerageHRIS.Enumeration
{
    public enum BloodGroup
    {
        [Display(Name = "A+")]
        APositive,
        [Display(Name = "A-")]
        ANegative,
        [Display(Name = "B+")]
        BPositive,
        [Display(Name = "B-")]
        BNegative,
        [Display(Name = "AB+")]
        ABPositive,
        [Display(Name = "AB-")]
        ABNegative,
        [Display(Name = "O+")]
        OPositive,
        [Display(Name = "O-")]
        ONegative,
        [Display(Name = "Unknown")]
        Unknown
    }
}

File: lib/core/enums/blood_group.dart
// Represents blood groups.

import 'package:freezed_annotation/freezed_annotation.dart';

part 'blood_group.g.dart';

@JsonEnum(alwaysCreate: true)
enum BloodGroup {
  @JsonValue(0)
  aPositive,
  @JsonValue(1)
  aNegative,
  @JsonValue(2)
  bPositive,
  @JsonValue(3)
  bNegative,
  @JsonValue(4)
  abPositive,
  @JsonValue(5)
  abNegative,
  @JsonValue(6)
  oPositive,
  @JsonValue(7)
  oNegative,
  @JsonValue(8)
  unknown,
}
